Sources/SwiftMemcache/MemcachedValue.swift

```diff
@@ -0,0 +1,56 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-memcache-gsoc open source project
+//
+// Copyright (c) 2023 Apple Inc. and the swift-memcache-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-memcache-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+
+/// Protocol defining the requirements for a type that can be converted to a ByteBuffer for transmission to Memcached.
+public protocol MemcachedValue {
```

Comment:
We should actually use this protocol in a follow up PR so that we have `set<Value: MemcacheValue>` and `get<Value: MemcacheValue>`

Reply:
Sounds good this will be properly introduced in a separate PR!